server/src/queries.rs

```diff
@@ -227,6 +229,17 @@ pub async fn confirm_flight(
             .into_inner();
         let sys_time = SystemTime::now();
         info!("confirm_flight: Flight plan with draft id: {} inserted in to storage with permanent id:{}", &fp_id, &fp.id);
+        let compliance_res = compliance_client_wrapper
```

Comment:
Do we want to do this in response to a customer confirming their flight?

A couple of concerns:
1) In the case of shared rideshares, several people may "confirm" the same flight plan
2) It may take the FAA, etc. several minutes to process a flight plan, blocking on their response may not be ideal
3) Indecisive people thrashing around with confirming and then deleting plans

I thought we might do this asynchronously like:
- svc-scheduler has a background thread that periodically (every 30 seconds?) checks the current flight plan list
- svc-scheduler submits a flight plan to svc-compliance  if 1) it's within N minutes to takeoff and 2) hasn't been submitted yet
- svc-compliance receives the flight plan submission and confirms it received it
- svc-scheduler moves along
- svc-scheduler periodically queries svc-compliance about the status of the open requests it has made
    - if svc-compliance has heard back from the govt endpoint, it will return the govt approval receipt
- svc-scheduler updates svc-storage with the approval/cert ID/signature

This allows the customer to potentially cancel more than N minutes prior to the flight without anything being sent to the FAA

Same thing with obtaining flight release, maybe with shorter periodicity

Reply:
Another thought: monitoring of flight plans and submitting requests to the FAA could be a background thread of `svc-compliance` instead of `svc-scheduler`, and `svc-compliance` could write directly to `svc-storage`'s flight plan record

`svc-scheduler` could then check with `svc-storage` about the status of govt. approval when doing final checks for takeoff without needing a background thread chatting with svc-compliance.

Keeps the web less tangled as well, the scheduler maybe wouldn't need to talk to svc-compliance...

Reply:
Good thoughts,
svc-compliance running a cron job and trying to confirm all flights within N minutes of start time, sounds like the best idea. We can also block cancelling of flight in UI N minutes before the flight commences.

But assuming for R2 scheduler doesn't rely on any other confirmation than "customer confirming the flight", we can maybe leave it as is, just remove blocking of svc-compliance call - so that we continue with confirmation in svc-storage immediately and only log error in case compliance service errors.

I think for scheduler it doesn't really matter if confirmed flights are also compliance_confirmed - we have to assume that flight will happen for the purpose of blocking resources for a time window. And if gov refuses to approve, then we simply cancel the flight, and free time window can be used by next customer request (assuming now gov will approve)

If however we need to rely on compliance confirmed status to block resources, we need to call compliance ASAP (as it is now during `confirm_flight`) and deal with potential cancellations etc.

Reply:
I think we should block resources without govt. confirmation for now until we can confirm how long it takes for the govt. endpoint to reply

If we learn that it's < 10 s because they have some sort of automation on their end then that's great and we could theoretically let the customer know during their confirmation request if the flight is even possible.

If govt approval takes any longer than 10 s, then we may have a multi-step confirmation where the customer initially sees:
1) Your flight has been reserved and submitted to the FAA for approval. You should receive notice within X minutes.

Then later gets an update/push notification/email with:
2) The FAA has approved your flight! You're ready to go! Show up at the vertipad 10 minutes in advance etc.

Reply:
We can leave as is for R2 since the "government endpoint" returns immediately right now  :laughing: 

In R3 maybe we can code a one minute delay before compliance responds, and test what the flow is like to the customer. In that case we'd be moving this submission to a chron job in `svc-compliance`.